src/main/java/entity/data_structure/DataStore.java

```diff
@@ -0,0 +1,47 @@
+package entity.data_structure;
+
+/**
+ * An abstract data structure.
+ * @param <T> an object for the data structure.
+ */
+public interface DataStore<T> {
+    /**
+     * Adds element to data structure.
+     * @param element value to add to list
+     */
+    void add(T element);
+
+    /**
+     * Removes element from data structure.
+     * @param element element to remove.
+     */
+    void remove(T element);
+
+    /**
+     * Gets element from data structure.
+     * @param element element to get.
+     * @return returns the element.
+     */
+    T get(T element);
+
+    /**
+     * Removes and returns the element that has been removed.
+     * @param element element to pop.
+     * @return element that has been popped.
+     */
+    T pop(T element);
+
+    /**
+     * Returns boolean if an element exists in the data structure.
+     * @param element element to check if contained.
+     * @return Boolean value if contained or not.
+     */
+    Boolean contains(T element);
+
+    /**
+     * Returns size of data structure.
+     * @return integer size.
+     */
+    Integer size();
+
+}
```

Comment:
good job following clean architecture and decoupling with interface + implementation.